### PR TITLE
[WIP] Change user for installing things

### PR DESCRIPTION
**Pull request by AlanCoding**
Depends-On: https://github.com/ansible/ansible-runner/pull/617

### CODE REVIEW

**Comment by AlanCoding** [comment]
Containerfile from building locally:

```
ARG ANSIBLE_RUNNER_IMAGE=quay.io/ansible/ansible-runner:devel
ARG PYTHON_BUILDER_IMAGE=quay.io/ansible/python-builder:latest

FROM $ANSIBLE_RUNNER_IMAGE as galaxy

USER root

ARG ANSIBLE_GALAXY_CLI_COLLECTION_OPTS=
ADD _build /build

WORKDIR /build
RUN ansible-galaxy role install -r requirements.yml --roles-path /usr/share/ansible/roles
RUN ansible-galaxy collection install $ANSIBLE_GALAXY_CLI_COLLECTION_OPTS -r requirements.yml --collections-path /usr/share/ansible/collections

FROM $PYTHON_BUILDER_IMAGE as builder
ADD _build/requirements_combined.txt /tmp/src/requirements.txt
ADD _build/bindep_combined.txt /tmp/src/bindep.txt
RUN assemble

FROM $ANSIBLE_RUNNER_IMAGE
USER root


COPY --from=galaxy /usr/share/ansible /usr/share/ansible

COPY --from=builder /output/ /output/
RUN /output/install-from-bindep && rm -rf /output/wheels
RUN alternatives --set python /usr/bin/python3
```

File: ansible_builder/main.py [hunk 1]
```diff
@@ -325,6 +325,8 @@ def __init__(self, definition,
             ),
             "",
             "FROM $ANSIBLE_RUNNER_IMAGE as galaxy",
+            "",
+            "USER root",
```
Comment (pabelanger) [inline ansible_builder/main.py:329]: have we thought of just using sudo, for our install commands?  Then, if we want we can remove the sudoers entry for the user or explain how someone can.

Comment (AlanCoding) [reply ansible_builder/main.py:329]: We can try it, we would need to document that all prepend / append steps will need to do the same. This PR should also need some changes to make those continue working.

**Comment by AlanCoding** [comment]
recheck

**Comment by AlanCoding** [comment]
This needs _so_ much more work.

**Comment by AlanCoding** [comment]
did this somewhere else

**Pull request closed without merging**